src/Core/Scheduler/Scheduler.js

```diff
@@ -256,6 +274,29 @@ Scheduler.prototype.getProtocolProvider = function getProtocolProvider(protocol)
     return this.providers[protocol];
 };
 
+
+Scheduler.prototype.addFormatParser = function addFormatParser(parser) {
+    // eslint-disable-next-line no-console
+    console.log(`Registering format : ${parser.format}`);
+    if (typeof (parser.parse) !== 'function') {
+        throw new Error(`Can't add parser for ${parser.format}: missing a parse function.`);
+    }
+    var that = this;
+    function register(format) {
+        that.parsers[format] = that.parsers[format] || [];
+        that.parsers[format].push(parser);
+    }
+    register(parser.format);
+    parser.extensions.forEach(register);
```

Comment:
I think it should be simple and only rely on `format`. We ditched `mimetypes` in #597, and I fail to see how we could benefit by having `extensions` and `mimetypes` on top of `format`.

Reply:
> extension may be the only thing you get when you are handed over a file (by url or drag n drop)

Extension is not really reliable, and can be confusing: a `xml` file can be a KML, GPX or even something else.

> mimetype is needed by wfs (in the query string)

It isn't really a mimetype, see for example the (usecase in GeoServer)[http://docs.geoserver.org/stable/en/user/services/wfs/outputformats.html]. But I see the necessity here.

> mimetype may be provided in the response header of fetches

Yes, but then we should maybe rework on the whole format/mimetype thing.

> format is for me the name of the parser given by itowns, with more uniform naming conventions

Agreed, that's why I think it should only be registered with the `format` option.

Reply:
I'm with @zarov : I'd rather not have 3 different ways of declaring a format.

What we could have though, is a format detector: is the format isn't explicitely specified, we can try to deduce from: url, filename, magic bytes, etc (something similar to what is done in [3dTileProvider](https://github.com/iTowns/itowns/blob/master/src/Provider/3dTilesProvider.js#L180) and [RasterProvider](https://github.com/iTowns/itowns/blob/master/src/Provider/RasterProvider.js#L79))